packages/tailwindcss/src/ast.ts

```diff
@@ -22,7 +22,12 @@ export type Context = {
   nodes: AstNode[]
 }
 
-export type AstNode = Rule | Declaration | Comment | Context
+export type AtRoot = {
+  kind: 'at-root'
```

Comment:
If we ever need more like this, we could have a more generic `marker` node where you attach an additional sub-kind of w/e. I think this is the best solution for now though.